admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -359,15 +359,33 @@ var featuresCollection = (function() {
       /** On feature selector event change, refresh possible values list */
       $(document).on('change', '.feature-collection select.feature-selector', function() {
         var selector = $(this).parent().parent().parent().find('.feature-value-selector');
```

Comment:
Is `$('.feature-collection select.feature-selector').parents('.row').find('.feature-value-selector')` give you the expected result?
You may add a more precise class to the `row` into the template and use it instead.